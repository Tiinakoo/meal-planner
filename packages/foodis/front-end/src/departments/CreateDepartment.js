import React from "react";
import TextInput from "../components/shared/TextInput/TextInput";
import { observer } from "mobx-react-lite";
import InputWrapper from "../components/private/InputWrapper/InputWrapper";
import translations from "../translations";
import getLanguage from "../doings/getLanguage/getLanguage";
import PrimaryButton from "../components/shared/PrimaryButton/PrimaryButton";
import Flex from "../components/shared/Flex/Flex";

const localTranslation = translations[getLanguage()];

const CreateDepartment = ({
  createDepartmentModel: { submit, name, isValid },
}) => {
  return (
    <form>
      <TextInput
        model={name}
        label={localTranslation.addNewDepartment}
        placeholder={localTranslation.placeHolderForTextInput}
        translation={localTranslation}
        data-department-name-input-test
      />

      <Flex alignRight>
        <PrimaryButton
          onClick={submit}
          size="sm"
          enabled={isValid}
          data-department-submit-test
        >
          {localTranslation.submit}
        </PrimaryButton>
      </Flex>
    </form>
  );
};

export default observer(CreateDepartment);
