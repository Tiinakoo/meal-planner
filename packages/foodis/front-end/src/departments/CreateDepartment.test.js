import CreateDepartment from "./CreateDepartment";
import React from "react";
import { mount } from "enzyme";

describe("CreateDepartment", () => {
  let component;
  let createDepartmentModelStub;

  beforeEach(() => {
    createDepartmentModelStub = {
      name: {},
      submit: () => {},
      isValid: {},
    };

    component = mount(
      <CreateDepartment createDepartmentModel={createDepartmentModelStub} />
    );
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });

  it("a name input within, has model", () => {
    const input = component.find("TextInput[data-department-name-input-test]");

    expect(input).toHaveProp("model", createDepartmentModelStub.name);
  });

  describe("a button within", () => {
    let button;

    beforeEach(() => {
      button = component.find("Button[data-department-submit-test]");
    });

    it("knows how to submit", () => {
      expect(button).toHaveProp("onClick", createDepartmentModelStub.submit);
    });

    it("knows when to be enabled", () => {
      expect(button).toHaveProp("enabled", createDepartmentModelStub.isValid);
    });
  });
});
