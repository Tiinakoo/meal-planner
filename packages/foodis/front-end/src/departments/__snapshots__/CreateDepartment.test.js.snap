// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDepartment renders 1`] = `
<CreateDepartment
  createDepartmentModel={
    Object {
      "isValid": Object {},
      "name": Object {},
      "submit": [Function],
    }
  }
>
  <form>
    <TextInput
      data-department-name-input-test={true}
      label="Add new department"
      model={Object {}}
      placeholder="Write here"
      translation={
        Object {
          "addNewDepartment": "Add new department",
          "placeHolderForTextInput": "Write here",
          "submit": "Submit",
        }
      }
    >
      <InputWrapper
        label="Add new department"
      >
        <div>
          <div>
            <BodyText
              font="primary"
              size="md"
            >
              <p
                className=""
              >
                Add new department
              </p>
            </BodyText>
          </div>
          <input
            data-department-name-input-test={true}
            onChange={[Function]}
            placeholder="Write here"
            translation={
              Object {
                "addNewDepartment": "Add new department",
                "placeHolderForTextInput": "Write here",
                "submit": "Submit",
              }
            }
            type="text"
          />
        </div>
      </InputWrapper>
    </TextInput>
    <Flex
      alignRight={true}
    >
      <div
        className=""
      >
        <PrimaryButton
          data-department-submit-test={true}
          enabled={Object {}}
          onClick={[Function]}
          size="sm"
        >
          <Button
            data-department-submit-test={true}
            enabled={Object {}}
            onClick={[Function]}
            size="sm"
            variant="primary"
          >
            <button
              className=""
              data-department-submit-test={true}
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <BodyText
                font="primary"
                size="md"
              >
                <p
                  className=""
                >
                  Submit
                </p>
              </BodyText>
            </button>
          </Button>
        </PrimaryButton>
      </div>
    </Flex>
  </form>
</CreateDepartment>
`;
